Update cake build script for new tooling

slang25 opened the pull request.

JosephWoodward commented on build.cake:
```diff
@@ -15,67 +18,90 @@ Task("Clean")
+                AppVeyor.UploadTestResults("./src/Shouldly.Tests/report.xml", AppVeyorTestResultsType.XUnit);
+        } catch(Exception)
+        {
+
```
Do we want to swallow this?

slang25 replied:
Yeah this is odd, was just doing this to debug. So it's actually throwing here, there is a `EnsureSuccessStatusCode` in the cake code, however the upload seems to work in that the tests all show up.

I'd like to know what is happening here, I'll do some more investigating.

JosephWoodward commented:
Looks like it's passing now, good job 👍 Just the one comment.

slang25 commented:
Sorry @JosephWoodward, only just noticed you had also started this, this was mostly from last night.

JosephWoodward commented:
No worries, hadn't got very far. Wanted to have a play around with Cake anyway so it's all good.

slang25 commented on build.cake:
```diff
@@ -15,67 +18,90 @@ Task("Clean")
+        msBuildSettings = new DotNetCoreMSBuildSettings()
+                            .WithProperty("Version", versionInfo.NuGetVersion)
+                            .WithProperty("AssemblyVersion", versionInfo.AssemblySemVer)
+                            .WithProperty("FileVersion", versionInfo.NuGetVersion);
```
What should FileVersion actually be?
Look in the build output at `Adding Environment Variable. name='GitVersion_` for a nice list of options

slang25 commented on src/Shouldly/Shouldly.csproj:
```diff
@@ -3,22 +3,18 @@
+    <IncludeSource>true</IncludeSource>
     <AssemblyOriginatorKeyFile>sn.snk</AssemblyOriginatorKeyFile>
     <SignAssembly>true</SignAssembly>
+    <PublicSign Condition=" '$(OS)' != 'Windows_NT' ">true</PublicSign>
```
All the cool kids have this in their repo, I have no idea what it does... shhh

JosephWoodward replied:
lol, I did wonder what it was.

slang25 commented on src/Shouldly/Shouldly.csproj:
```diff
@@ -3,22 +3,18 @@
     <PackageLicenseUrl>https://github.com/shouldly/shouldly/blob/master/LICENSE.txt</PackageLicenseUrl>
-    <NetStandardImplicitPackageVersion>1.6.1</NetStandardImplicitPackageVersion>
-    
-    <GeneratePackageOnBuild>True</GeneratePackageOnBuild>
```
Took out `GeneratePackageOnBuild`, on macOS you can do a typical `dotnet build`, and if you want a package then run `.\build.sh`

slang25 commented on src/Shouldly/Shouldly.csproj:
```diff
@@ -3,22 +3,18 @@
     <PackageIconUrl>https://raw.github.com/shouldly/shouldly/master/Icons/package_icon.png</PackageIconUrl>
     <PackageProjectUrl>http://shouldly.github.com</PackageProjectUrl>
     <PackageLicenseUrl>https://github.com/shouldly/shouldly/blob/master/LICENSE.txt</PackageLicenseUrl>
-    <NetStandardImplicitPackageVersion>1.6.1</NetStandardImplicitPackageVersion>
```
Took out this, cause apparently that's best practice

JosephWoodward replied:
Where did you read that?

slang25 replied:
Haven't found the source of that remark.
I did find this:
https://github.com/shouldly/shouldly/pull/432#discussion_r125122510
That was mostly a objection to use `1.6.0` for the most part, I believe we get whatever the default for the sdk is, for 1.0.4 sdk = 1.6.1 implicit netstandard package. I think we should let it be implicit unless we understand a reason to specify it in order to not trip ourselves up in future.

slang25 commented:
@JosephWoodward I've addressed the comment, we didn't need to upload the results file as AppVeyor does this for us somehow by magic.

Just need to make a decision on the `FileVersion` and we are good to go.

slang25 commented:
@JosephWoodward this one is still ready to go, if you wanted a version out tonight this would be a good chance to test the deployment via AppVeyor.
